src/book/02-system/01-nubots/02-build-system.mdx

```diff
@@ -272,6 +272,62 @@ Be sure that you are using the `generic` target for this (`./b target generic`),
 
 </Alert>
 
+The `run` command provides some options to help you debug your roles.
+
+<Alert type="warning">
+
+Be sure that you are using the `generic` target for this (`./b target generic`). You will likely want to recompile your role after setting `CMAKE_BUILD_TYPE` to either `Debug` or `RelWithDebInfo` otherwise you will likely get useless information.
+
+</Alert>
+
+
+| Subcommand     | Description                                                                              |
+| -------------- | ---------------------------------------------------------------------------------------- |
+| --asan         | Enables address sanitisation, a fast memory error detector.                              |
+| --gdb          | Runs selected role inside of GDB to enable debugging.                                    |
+| --valgrind     | Runs selected role inside of valgrind to enable detection of memory errors.              |
+
+In order to use ASan you must recompile your role after setting `USE_ASAN` to `ON` in the cmake options (`./b configure -i`).
+
+Currently, using the `--asan` option adds an environment variable that will direct ASan to log all of its output to a file. This is done to ensure that even when you are running a curses-enabled role (e.g. `keyboardwalk`) you will always be able to revocer the output from ASan (and it will also be clean from other log messages). There are a number of other run-time options you could set to tune the behaviour of ASan, they can be found [here](https://github.com/google/sanitizers/wiki/AddressSanitizerFlags). Just run your role as follows
+
+```sh
+ASAN_OPTIONS=option1=value1:option2=value ./b run --asan keyboardwalk
+```
+
+<Alert type="info">
+
+If you would like to use a different path for your ASan log file be sure to specify `ASAN_OPTIONS=log_path=path/to/log/file`. However, there are very few locations inside of docker that have persistent write access.
+
+</Alert>
+
+ASan and GDB can be run together. For more details on how this works have a look at [this page](https://github.com/google/sanitizers/wiki/AddressSanitizerAndDebugger).
+
+Since ASan and valgrind perform the same task they cannot be used in conjunction and an exception will be thrown if you try to do so.
+
+While it is possible to run GDB and valgrind simultaneously, two separate processes need to be started and then linked together. While this is technically possible, it has currently been assigned to the `too hard` basket and, as such, this option has been disabled and an exception will be thrown if you try to do this.
+
+<Alert type="info">
```

Comment:
Something about these three alerts in a row feels weird. Perhaps just have the content in a list?